### DISCUSSION TITLE
Static export, no server actions, what gives?

### DISCUSSION THREAD

**Discussion by EricWVGG** [vercel/next.js#73559] [App Router]
I’m trying to get a site with `output: export`, and am getting the error `Server Actions are not supported with static export.`

A build _without_ static explicitly defined shows no dynamic or ISR routes:
```
Route (app)                              Size     First Load JS
┌ ○ /                                    254 B           176 kB
├ ○ /_not-found                          901 B           101 kB
├ ● /[projectType]                       254 B           176 kB
├   ├ /lighting
├   ├ /collaboration
├   └ /objects
└ ● /[projectType]/[slug]                254 B           176 kB
    ├ /lighting/lamp
    ├ /collaboration/thing
    └ /objects/fork
...
○  (Static)  prerendered as static content
●  (SSG)     prerendered as static HTML (uses generateStaticParams)
```

There's no declaration of `use server` anywhere, and I've been over the ["unsupported features" checklist](https://nextjs.org/docs/app/building-your-application/deploying/static-exports#unsupported-features). No API routes. Custom image loader. Any guesses? (and btw shouldn't this error indicate where it's seeing this issue?)

Next 15.0.3. help me obi wan etc etc

**Comment by icyJoseph** [comment]
Hi,

Perhaps a 3rd party dependency uses server actions? Got a link to a repository?

**Reply by EricWVGG** [reply]
sure

https://github.com/EricWVGG/farrah-sit

**Reply by icyJoseph** [reply]
Yeah, it was a 3rd party: https://github.com/sanity-io/next-sanity/issues/1899

- https://github.com/sanity-io/next-sanity/blob/ce7fbfc0238a70a06c38233cc17c14aa3dd21ab4/packages/next-sanity/src/visual-editing/server-actions/index.ts#L1

Follow up the issue on the `next-sanity` repository: https://github.com/sanity-io/next-sanity/issues/1899

**Reply by Dagwal** [reply]
yeah this maybe

**Comment by aaron-denari** [comment]
I have also experienced this problem and it's extremely frustrating.  It is mind blowing that this error doesn't tell you exactly where the problem is!

**Comment by doppelgunner** [comment]
You can try this approach, this worked for me:
- https://www.nxgntools.com/blog/improve-nextjs-app-page-load-with-server-actions-and-nextauth
